extensions/cornerstone/src/services/ViewportService/CornerstoneViewportService.ts

```diff
@@ -548,115 +568,153 @@ class CornerstoneViewportService extends PubSubService
     const displaySetInstanceUIDs = this.viewportsDisplaySets.get(viewport.id);
 
     const segDisplaySet = displaySetInstanceUIDs
-      .map(displaySetService.getDisplaySetByUID)
+      .map(DisplaySetService.getDisplaySetByUID)
       .find(displaySet => displaySet && displaySet.Modality === 'SEG');
 
-    if (segDisplaySet) {
-      const { referencedVolumeId } = segDisplaySet;
-      const referencedVolume = cache.getVolume(referencedVolumeId);
-      const segmentationId = segDisplaySet.displaySetInstanceUID;
-
-      const toolGroup = toolGroupService.getToolGroupForViewport(viewport.id);
+    const rtssDisplaySet = displaySetInstanceUIDs
+      .map(DisplaySetService.getDisplaySetByUID)
+      .find(displaySet => displaySet && displaySet.Modality === 'RTSTRUCT');
```

Comment:
This find assumes there is 0 or 1 instance - would rather have a single unified map as suggested above that includes all rt and segs and adds them all, based on the display set attribute that indicates how to handle it.  Then there is ONE addOverlayRepresentationForDisplaySet which adds for any time of segmentation.